src/Components/Facility/Consultations/NursingPlot.tsx

```diff
@@ -71,6 +71,14 @@ export const NursingPlot = ({ consultationId }: any) => {
     else return false;
   };
 
+  const groupedByDate: any = {};
+  dataToDisplay.forEach((item: any) => {
+    if (!groupedByDate[item.date]) {
+      groupedByDate[item.date] = [];
+    }
+    groupedByDate[item.date].push(item);
+  });
```

Comment:
Can we keep the logic too similar to what's present in the Routine table at least? 

Ideally let's have a reusable component for this, as it could be re-used in couple of other places as well eventually.

Let's also avoid the usage of `any`

Reply:
Hey @rithviknishad,  I tired to make a reusable component but the model of nursing care procedures is different and I am not able to extract data in the same way as in Routine , could you please guide me how can I proceed.

Reply:
You can pass the list of fields as a prop

Reply:
hey @rithviknishad , the Routine section has fields and subfields but the NURSING_CARE_PROCEDURES have only field , so using the same component is not good. I cat still use some conditions but that is not looking Ideal. Should I make the component?

Reply:
you can simply skip defining sub-fields for procedures right?

Reply:
Hey @rithviknishad  I made a new PR with the changes . https://github.com/ohcnetwork/care_fe/pull/8608 .